File: tks/WebService/src/main/kotlin/webservice/webservices/LibraryItemWebService.java
package webservice.webservices;

import domain.exceptions.ResourceNotFoundException;
import domain.exceptions.TypeValidationFailedException;
import webservice.adapters.LibraryItemWebServiceAdapter;
import webservice.dto.LibraryItemSoapDto;

import javax.inject.Inject;
import javax.jws.WebMethod;
import javax.jws.WebService;

@WebService
public class LibraryItemWebService {

    @Inject
    private LibraryItemWebServiceAdapter serviceAdapter;

    @WebMethod
    public LibraryItemSoapDto getLibraryItem(String accessionNumber) throws TypeValidationFailedException, ResourceNotFoundException {
        try{
            return serviceAdapter.getLibraryItem(accessionNumber);
        } catch (TypeValidationFailedException | ResourceNotFoundException e) {
            throw e;
        }
    }
}

File: tks/WebService/src/main/kotlin/webservice/webservices/LibraryItemWebService.kt
package webservice.webservices

import domain.exceptions.ResourceNotFoundException
import domain.exceptions.TypeValidationFailedException
import webservice.adapters.LibraryItemWebServiceAdapter
import webservice.dto.LibraryItemSoapDto
import javax.inject.Inject
import javax.jws.WebMethod
import javax.jws.WebService

@WebService(targetNamespace = "http://superbiz.org/wsdl")
interface ILibraryItemWebService {
    @WebMethod
    @Throws(TypeValidationFailedException::class, ResourceNotFoundException::class)
    fun getLibraryItem(accessionNumber: String): LibraryItemSoapDto
}

@WebService(
    portName = "LibraryItemPort",
    serviceName = "LibraryItemService",
    targetNamespace = "http://superbiz.org/wsdl",
    endpointInterface = "webservice.webservices.ILibraryItemWebService")
open class LibraryItemWebService : ILibraryItemWebService {
    @Inject
    private var serviceAdapter: LibraryItemWebServiceAdapter? = null
    @WebMethod
    @Throws(TypeValidationFailedException::class, ResourceNotFoundException::class)
    override fun getLibraryItem(accessionNumber: String): LibraryItemSoapDto {
        return try {
            serviceAdapter!!.getLibraryItem(accessionNumber)
        } catch (e: TypeValidationFailedException) {
            throw e
        } catch (e: ResourceNotFoundException) {
            throw e
        }
    }
}